src/manifolds/Symplectic.jl

```diff
@@ -0,0 +1,889 @@
+@doc raw"""
+    Symplectic{n, ℝ} <: AbstractEmbeddedManifold{ℝ, DefaultIsometricEmbeddingType}
+
+Over the real number field ℝ the elements of the symplectic manifold
+are all $2n × 2n$ matrices satisfying the requirement
+````math
+\operatorname{Sp}(2n, ℝ) = \bigl\{ p ∈ ℝ^{2n × 2n} \, \big| \, p^TQ_{2n}p = Q_{2n} \bigr\},
+````
+where
+````math
+Q_{2n} =
+\begin{bmatrix}
+  0_n & I_n \\
+ -I_n & 0_n
+\end{bmatrix},
+````
+with $0_n$ and $I_n$ denoting the $n × n$ zero-matrix
+and indentity matrix in ``ℝ^{n \times n}`` respectively.
+
+Internally the dimensionality of the structure is stored as half of the even dimension
+supplied to the constructor, `Symplectic(2n) -> Symplectic{n}()`,
+as most computations with points on the Real Symplectic manifold takes advantage of the
+natural block structure
+of a matrix ``A ∈ ℝ^{2n × 2n}`` where we consider it as consisting of four
+smaller matrices in ``ℝ^{n × n}``.
+"""
+struct Symplectic{n,𝔽} <: AbstractEmbeddedManifold{𝔽,DefaultIsometricEmbeddingType} end
+
+@doc """
+    Symplectic(n, field::AbstractNumbers=ℝ) -> Symplectic{div(n, 2), ℝ}()
+
+# Constructor:
+The constructor for the [`Symplectic`](@ref) manifold accepts the even embedding
+dimension ``n = 2k`` for the real symplectic manifold, ``ℝ^{2k × 2k}``.
+"""
+function Symplectic(n::Int, field::AbstractNumbers=ℝ)
+    n % 2 == 0 || throw(ArgumentError("The dimensionality of the symplectic manifold
+                        embedding space must be even. Was odd, n % 2 == $(n % 2)."))
+    return Symplectic{div(n, 2),field}()
+end
+
+@doc raw"""
+The canonical Riemannian metric on the symplectic manifold,
+defined pointwise for ``p \in \operatorname{Sp}(2n)`` by [^FioriSimone2011]
+````math
+\begin{align*}
+    & g_p \colon T_p\operatorname{Sp}(2n) \times T_p\operatorname{Sp}(2n) \rightarrow ℝ, \\
+    & g_p(Z_1, Z_2) = \operatorname{tr}((p^{-1}Z_1)^T (p^{-1}Z_2)).
+\end{align*}
+````
+This metric is also the default metric used within `Maniolds.jl`.
+"""
+struct RealSymplecticMetric <: RiemannianMetric end
+
+struct ExtendedSymplecticMetric <: AbstractMetric end
+
+struct CayleyInverseRetraction <: AbstractInverseRetractionMethod end
+
+@doc raw"""
+    SymplecticMatrix{T}
+
+A lightweight structure to represent the action of the matrix
+representation of the canonical symplectic form,
+````math
+Q_{2n} =
+\begin{bmatrix}
+0_n & I_n \\
+ -I_n & 0_n
+\end{bmatrix} \quad \in ℝ^{2n \times 2n},
+````
+such that
+````math
+\omega_{2n}(x, y) = x^TQ_{2n}y, \quad x, y \in ℝ^{2n}.
+````
```

Comment:
Shouldn't λ appear somewhere here?

Reply:
I've tried to clarify the use of the scalar `λ` to keep track of scaling and transpose operations now.

Reply:
What I meant is that why
```
Q_{2n} =
\begin{bmatrix}
0_n & I_n \\
 -I_n & 0_n
\end{bmatrix} \quad \in ℝ^{2n \times 2n},
```
isn't something like
```
Q_{2n}(λ) =
\begin{bmatrix}
0_n & λI_n \\
 -λI_n & 0_n
\end{bmatrix} \quad \in ℝ^{2n \times 2n},
```
?

Reply:
Sorry, github showed me the old version for some reason when I was re-checking it.